Solutions/BrianCarName.java

```diff
@@ -0,0 +1,87 @@
+package com.core.programming;
+
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * The Class BrianCarName
+ *
+ */
+public class BrianCarName {
+	
+	/** The logger. */
+	private Logger logger;
+
+	/**
+	 * Instantiates a new BrianCarName.
+	 */
+	public BrianCarName() {
+		logger = Logger.getLogger(BrianCarName.class.getName());
+	}
+	
+	/**
+	 * Checks if given car name is liked by Brian
+	 * 
+	 * @param carName the given car name
+	 * @return the string
+	 */
+	public String checkCarName(String carName) {
+		if (carName != null  && !carName.isEmpty() && !carName.trim().isEmpty()) {
+			logger.log(Level.INFO, "Input String/Car Name : " + carName);
+			TreeMap<Character,Integer> charOccurrenceMap = populateMap(carName);
```

Comment:
@neerajkujha Yes TreeMap is required, since I'm using `charOccurrenceMap.firstEntry()` further in the solution. i.e `firstEntry` api is available only for TreeMap but not in case of HashMap or LinkedHashMap. But then yes, solution is possible by using HashMap as well, in that case i would need to convert values of map to an array & get its 1st element to achieve same as above.

Reply:
Good, but treemap keeps key in sorted order, do you think we need the keys in order here. Also do we really need firstentry

Reply:
The only purpose I've used `TreeMap` here is for `firstEntry` api. The reason being I could stop validating further for character (C3) occurrence, in a scenario where 2nd character (C2) occurrence is not equal to 1st character (C1) occurrence. Yes I agree, we could use other simpler approach like putting C1, C2, C3 occurrences in `HashSet` & validating if it's size is equal to one. I had chosen former approach thinking we could stop validating an iteration before, in the above scenario.